app/src/backend/environment/hooks/useEnvironmentManager.ts

```diff
@@ -215,6 +215,27 @@ const useEnvironmentManager = (initListenerAndFetcher: boolean = false) => {
     [allEnvironmentData]
   );
 
+  const getVariableData = useCallback(
+    (variableName: string) => {
+      if (allEnvironmentData[currentEnvironmentId].variables[variableName]) {
+        return {
+          ...allEnvironmentData[currentEnvironmentId].variables[variableName],
+          name: variableName,
+        };
+      } else {
+        // returns the variable data from the first environment where it is present
```

Comment:
When will this case happen?

Reply:
If the variable used is not present in the current active environment then we look for the value in other envs

Reply:
Updated. Postman as well does not add vars from other env